### ISSUE TITLE
"Cannot read property 'slice' of undefined" when parsing response

### ISSUE THREAD

**Issue by cmstolper** [adopted-ember-addons/ember-file-upload#205]
Sometimes, when uploading and parsing the response, the "Cannot read property 'slice' of undefined" error occurs at `ember-file-upload/addon/system/http-request.js::L26` of the `parseResponse` method.

https://github.com/adopted-ember-addons/ember-file-upload/blob/750295f49f46a81295ff35d612dedecb5357a594/addon/system/http-request.js#L23-L29

I believe it is because even though `parts` may pass `parts.length > 0`, it may not have a value at `[1]` or `[2]`.

**Comment by Alonski** [comment]
@cmstolper Could you please try to create a Twiddle or Repo recreation of this issue?

**Comment by cmstolper** [comment]
Gladly if I get the chance, but in the meantime I wanted to make sure I pointed out the potential for bugs in general at this point:
```
if (parts.length > 0){ 
    E[parts[1].slice(0, -1)] = trim(parts[2]); 
}
```
...since `parts` could have a length of 1, which would pass the if statement, but error when trying to check the positions at `[1]` or `[2]`.

**Comment by happydude** [comment]
I think the problem is the regex is too restrictive, if the header field-name has a character outside of [0-9A-Za-z_-]*: then only parts[0] will exist. According to the RFC all printable ascii characters except for the colon are valid characters.

See RFC2616 section 4.2 followed by RFC822

**Comment by gilest** [comment]
@Alonski I think this can be closed.

**Comment by jelhan** [comment]
Closing this one as the main issue seems to be fixed by #254.

I'm not sure if the two strict regexp mentioned by @happydude two comments above is a real-world issue. Feel free to reopen if so.

**Comment by happydude** [comment]
the header that caused issue for us was one with ' in it
```
< HTTP/1.1 503 Service Unavailable
< Date: Thu, 23 Jan 2020 23:57:40 GMT
< Server: Apache/2.4.6 (Red Hat Enterprise Linux) PHP/7.2.24
< Strict-Transport-Security: max-age=63072000; includeSubDomains
< X-Powered-By: PHP/7.2.24
< 'X-Powered-By: ZendServer/2019.0.3'
< ZRay-ID: 3@119304@1579022073@0
```

But it is moot now
